### PR TITLE
@izakp => Basic Sentry support

### PR DESCRIPTION
**Pull request by mzikherman**
This adds basic Sentry support, which should be safe to merge before setting ENV vars.

I created `metaphysics-staging`, and locally, the output looks pretty nice:

screen shot 2018-03-07 at 3 40 32 pm

Decided to PR this to your `hokusai` branch since that's close to being rolled out. But I can maybe open this against `master` if we want to try this out sooner.

cc @alloy -> I'm sure there's a lot I'm missing/gotchas for production (source maps, etc.). I'm thinking that even with that stuff not working, this might be mergable, tho I am planning on looking at that next (pointers welcome!)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "engines": {
     "node": "8.4.0",
     "npm": "5.3.0",
-    "yarn": "1.0.1"
+    "yarn": "1.*"
```
Comment (mzikherman) [inline package.json]: I moved this since this prevents `yarn install` from running (sans Docker/CI) which might throw some people off.

**Comment by alloy** [comment]
Yeah the reason it works fine is probably because you were locally testing the JIT compiled dev code rather than the production AOT compiled build? The emitted code in dev mode keeps artefacts such as source-maps in memory.

I think your strategy of _something_ now and improving later is fine, though 👍 Let me know if you need a rubber ducky for when you wanna make source-maps work.

**Comment by mzikherman** [comment]
Ah, let me experiment with running the babel compiled version locally and see how that output looks.

**Comment by alloy** [comment]
Oh, actually [it looks](https://github.com/artsy/metaphysics/blob/master/package.json#L23) like we’re compiling source-maps inline, so it will probably still work, but it’s not the best situation.

**Comment by izakp** [comment]
You can test with `hokusai dev start` which will [run the babel compiler in the Docker build](https://github.com/artsy/metaphysics/blob/hokusai/Dockerfile#L25)

**Comment by izakp** [comment]
This is an example of why I'd ideally like to keep the hokusai / docker dev environment as production-like as possible without including tooling (i.e. hot-module reloading and JIT compilation) pertinent to local dev setup - this way we are able to check a production-like build against our loacl Docker environment quickly and easily cc @damassi

File: package.json (hunk 1 above)
Comment (izakp) [reply package.json]: Cool thanks - I forgot to loosen that dependency.  It's pinned in the Dockerfile so should be consistent on CI at least.

File: src/index.js [hunk 2]
```diff
@@ -24,11 +25,13 @@ const {
   ENABLE_SCHEMA_STITCHING,
   NODE_ENV,
   QUERY_DEPTH_LIMIT,
+  SENTRY_PRIVATE_DSN,
 } = config
 const isProduction = NODE_ENV === "production"
 const queryLimit = (QUERY_DEPTH_LIMIT && parseInt(QUERY_DEPTH_LIMIT, 10)) || 10 // Default to ten.
 const enableSchemaStitching = ENABLE_SCHEMA_STITCHING === "true"
 const enableQueryTracing = ENABLE_QUERY_TRACING === "true"
+const enableSentry = !!SENTRY_PRIVATE_DSN
```
Comment (izakp) [inline src/index.js]: Oh yea forgot all about that `!!` ;)

**Comment by izakp** [comment]
@mzikherman I can't find the Sentry DSN in http://sentry.io/artsynet/ - maybe I need to be an admin to see it?

Can you set `SENTRY_PRIVATE_DSN` for both staging and production?  Otherwise LGTM.

**Comment by mzikherman** [comment]
@izakp for reference: you have to select a specific project from the dropdown, and then click 'Settings', at which point you should see 'Client Keys (DSN)' in the sidebar.

You do have to be logged in as an owner (it@) to see 'Settings' however.

**Comment by mzikherman** [comment]
ENV set in Hokusai, merging!

**Comment by damassi** [comment]
> This is an example of why I'd ideally like to keep the hokusai / docker dev environment as production-like as possible without including tooling (i.e. hot-module reloading and JIT compilation) pertinent to local dev setup - this way we are able to check a production-like build against our loacl Docker environment quickly and easily

When looking at the overall complexity of MP (which is very low), this isn't a good compromise compared to development speed. JIT compilation is a fact of life in Node.js during development, and express-reloadable increases development speed my many multiples.

I don't see why we can't a) rapid iteration times in a common Node.js development environment; and b) support Docker by addressing the problem? If the issue is with inline sourcemaps -- `-s inline` (not a prod environment)` -- a [quick look at the docs](https://babeljs.io/docs/usage/cli/#compile-with-source-maps) reveals that there is an option to output `.map.js` files. If the issue is getting super accurate sourcemaps into Sentry, that's a whole different challenge as there are lots of configuration steps involved there completely unrelated to Docker.

**Comment by izakp** [comment]
@damassi yes I mean that we should support both, as you say :)

**Comment by mzikherman** [comment]
@damassi yea I think the inline sourcemap isn't a Docker issue or anything (it's just related to a 'larger' issue of potential differences between local dev, docker, staging/prod). I think those dev issues remain the express reloadable and `yarn link` issues mentioned earlier.

Re: inline source maps, I'm not actually 100% sure why they are a bad thing, except for possibly defeating the purpose of minification, etc. And in that case yes, we could definitely generate/upload them separately during CI, and we could point Sentry to that.

However, since Metaphysics doesn't actually run in your browser, I'm wondering if it's actually that bad to just inline source maps in prod?

**Comment by damassi** [comment]
@mzikherman - Inline source-maps can be pretty heavy filesize / memory-wise, but like you said since nothing is being sent down to the browser the impact is probably pretty small in our case. It is considered a bad practice though and does increase the memory footprint (sometimes considerably).

**Comment by izakp** [comment]
You can see memory usage for staging here: https://kubernetes-staging.artsy.net/#!/deployment/default/metaphysics-web?namespace=default and here https://app.datadoghq.com/containers?columns=container_name,container_cpu,container_memory,container_net_sent_bps,container_net_rcvd_bps,container_status,container_started&options=normalizeCPU&sort=container_memory,DESC&tags=env%3Astaging,kube_deployment%3Ametaphysics-web

**Comment by damassi** [comment]
Seems pretty small IMO, and since inline sourcemaps are already turned on maybe we should just monitor for now? (I'm curious to see what will happen if we triggered an error in prod)

**Pull request merged**